### ISSUE TITLE
sphinxsearch needs expat for xmlpipe2 support

### ISSUE THREAD

**Issue by vvs-** [NixOS/nixpkgs#70076]
**Describe the bug**
To enable xmlpipe2 support `expat` needs to be in nativeBuildInputs for `sphinxsearch` expression.

**To Reproduce**
1. Run `sphinxsearch-indexer` with xmlpipe2 source in configuration file.
2. Observe an error that XML support is missing.

**Expected behavior**
xmlpipe2 sources should work out of the box.

**Metadata**
```
 - system: `"i686-linux"`
 - host os: `Linux 4.19.72, NixOS, 19.03.173489.af5481e9b77 (Koi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.2.2`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by veprbl** [comment]
cc @ederoyd46

**Comment by vvs-** [comment]
Turns out that it also doesn't support MySQL. Here's a patch which solves both problems, though expat probably belongs to buildInputs:
```
--- default.nix~	1970-01-01 00:00:01.000000000 +0000
+++ default.nix	2020-01-31 23:18:35.840418572 +0000
@@ -1,4 +1,4 @@
-{ stdenv, fetchurl, pkgconfig,
+{ stdenv, fetchurl, pkgconfig, expat, mariadb, zlib, openssl,
   version ? "2.2.11",
   mainSrc ? fetchurl {
     url = "http://sphinxsearch.com/files/sphinx-${version}-release.tar.gz";
@@ -12,12 +12,17 @@
 
   configureFlags = [
     "--program-prefix=sphinxsearch-"
-    "--without-mysql"
+    "--with-mysql=${mariadb}/"
+    "--with-libexpat"
     "--enable-id64"
   ];
 
   nativeBuildInputs = [
-    pkgconfig
+    pkgconfig expat
+  ];
+
+  buildInputs = [
+    mariadb zlib openssl
   ];
 
   meta = {
```

**Comment by veprbl** [comment]
@vvs- Do you mind submitting a PR?

**Comment by vvs-** [comment]
Done: #79032 

Sorry, if I'm not up to PR expectations, but I believe these changes are tiny.

**Comment by jtojnar** [comment]
Fixed by #80727
